src/card_operations/card_internal.c

```diff
@@ -0,0 +1,370 @@
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "apdu.h"
+#include "app_error.h"
+#include "buzzer.h"
+#include "card_operations.h"
+#include "events.h"
+#include "nfc.h"
+#include "nfc_events.h"
+#include "ui_instruction.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define HANDLE_P0_EVENTS(p0_event)                                             \
+  do {                                                                         \
+    if (true == p0_event.inactivity_evt) {                                     \
+      return CARD_OPERATION_P0_TIMEOUT_OCCURED;                                \
+    } else if (true == p0_event.abort_evt) {                                   \
+      return CARD_OPERATION_P0_ABORT_OCCURED;                                  \
+    }                                                                          \
+  } while (0)
+
+#define NFC_SET_ERROR_MSG(card_data, msg)                                      \
+  (card_data->error_message =                                                  \
+       (card_data)->error_message ? (card_data)->error_message : (msg))
+#define NFC_GET_ERROR_MSG(card_data)                                           \
+  ((card_data)->error_message ? (card_data)->error_message : "(none)")
+
+#define NFC_RETURN_SUCCESS(card_data)                                          \
+  return card_data->error_type = CARD_OPERATION_SUCCESS
```

Comment:
please reconsider doing return and assignment in the same statement

Reply:
updated assignment and return in separate statements.